src/identity.cc

```diff
@@ -947,6 +1030,11 @@ TRITONBACKEND_ModelInstanceExecute(
           TRITONSERVER_LOG_VERBOSE,
           (std::string("\trequested_output ") + output_name).c_str());
 
+#ifdef TRITON_ENABLE_METRICS
+      GUARDED_RESPOND_IF_ERROR(
+          responses, r, model_state->UpdateMetrics(input_byte_size));
```

Comment:
I think you need a mutex in the `UpdateMetrics` function incase there are multiple instances of the model.

Reply:
@Tabrizian I believe it's OK without a mutex since these methods are just a passthrough to a [thread-safe](https://github.com/jupp0r/prometheus-cpp/blob/master/core/include/prometheus/counter.h#L24-L39) `Increment(value)` call. Please do correct me if I'm wrong though.

UpdateMetrics -> [TRITONSERVER_MetricIncrement](https://github.com/triton-inference-server/core/blob/0141e0651c4355bf8a9d1118aac45abda6569997/src/tritonserver.cc#L2873) -> [MetricFamily::Increment](https://github.com/triton-inference-server/core/blob/0141e0651c4355bf8a9d1118aac45abda6569997/src/metric_family.cc#L171) -> [prometheus::Counter::Increment](https://github.com/jupp0r/prometheus-cpp/blob/bac072291a029f09ce903b6757919f8111234b36/core/include/prometheus/counter.h#L24-L39)

Reply:
I see. Looks like the Prometheus API is thread-safe so a mutex is not required.